features/net/network-socket/SocketAddress.h

```diff
@@ -78,9 +78,10 @@ class SocketAddress {
    
     /** Set the IP address
      *
-     *  @param addr     Null-terminated represention of the IP address
+     *  @param addr     Null-terminated representation of the IP address
+     *  @return         True if address was a valid IP literal
      */
-    void set_ip_address(const char *addr);
+    bool set_ip_address(const char *addr);
```

Comment:
The SocketAddress will also be "null" after a valid parse of "0.0.0.0". Need to distinguish.